### PR TITLE
IonSpigot - Smooth & lag compensated potions

### PR DESCRIPTION
**Pull request by Rastrian**
- https://github.com/Samsuik/IonSpigot/blob/977187c2c4cbd96b0fbf589644a376b19d4f5930/TacoSpigot-Server-Patches/0025-Lag-Compensated-Potions.patch

### CODE REVIEW

**Comment by Lucaskyy** [comment]
Did you close it by accident? :P

**Comment by Lucaskyy** [comment]
@Rastrian Is this PR done?

**Review by Lucaskyy — APPROVED** [review]

File: NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java [hunk 1]
```diff
@@ -309,6 +309,20 @@ private static void hideProjectilesFromHiddenPlayers() {
         c.addComment("settings.hide-projectiles-from-hidden-players", "Hides projectiles from hidden players");
     }
 
+    public static boolean lagCompensatedPotions;
+    
+    private static void lagCompensatedPotions() {
+        lagCompensatedPotions = getBoolean("settings.lag-compensated-potions", false);
+        c.addComment("settings.lag-compensated-potions", "Enables lag compesation throwing potions");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java:316]: Bump config version

**Review by Elierrr — CHANGES_REQUESTED** [review]

**Comment by Rastrian** [comment]
> @Rastrian Is this PR done?

basically yeah, im maybe thinking in draft and do hit detection / knockback async

**Comment by CyberFlameGO** [comment]
That doesn't sound like a good idea

**Comment by andreasdc** [comment]
> > @Rastrian Is this PR done?

It will be the best to improve performance of ticking and movement of entities. Implementing Starlight would be nice too.

**Comment by Rastrian** [comment]
> > > @Rastrian Is this PR done?
> 
> It will be the best to improve performance of ticking and movement of entities. Implementing Starlight would be nice too.

im literally working with async entity tracking now

**Comment by Rastrian** [comment]
> That doesn't sound like a good idea

Agreed, maybe doing little PRs its better.

**Comment by andreasdc** [comment]
> w

Waste of time

**Comment by Rastrian** [comment]
> > w
> 
> Waste of time

ewwww, async entity tracking literally fix issues with entity movement.

**Comment by andreasdc** [comment]
> > > w
> > 
> > 
> > Waste of time
> 
> ewwww, async entity tracking literally fix issues with entity movement.

Ok then waste some time and come back when you will get to my point of view :P

**Comment by Rastrian** [comment]
> > > > w
> > > 
> > > 
> > > Waste of time
> > 
> > 
> > ewwww, async entity tracking literally fix issues with entity movement.
> 
> Ok then waste some time and come back when you will get to my point of view :P

what point of view dude? wtf are you talking about.

**Comment by andreasdc** [comment]
> > w
> 
> Waste of time

this + you won't optimize movement by doing async tracker, it's better to focus on improving code of movement and ticking

**Comment by Rastrian** [comment]
> > > w
> > 
> > 
> > Waste of time
> 
> this + you won't optimize movement by doing async tracker, it's better to focus on improving code of movement and ticking

an parallel entity tracking its literally optimizing the movement of entity.

IonSpigot implements the config faster-cannon-tracker, which improves a bit how entities work compared to the original implementation. But that doesn't totally solve the issue because this movement is intense when you put several entities simultaneously.

**Comment by Rastrian** [comment]
> > > w
> > 
> > 
> > Waste of time
> 
> this + you won't optimize movement by doing async tracker, it's better to focus on improving code of movement and ticking
> 
> I'm far forward in such things.

If you're really ahead why don't you just raise a PR contributing this issue?

**Comment by andreasdc** [comment]
> > > > w
> > > 
> > > 
> > > Waste of time
> > 
> > 
> > this + you won't optimize movement by doing async tracker, it's better to focus on improving code of movement and ticking
> 
> an parallel entity tracking its literally optimizing the movement of entity.
> 
> IonSpigot implements the config faster-cannon-tracker, which improves a bit how entities work compared to the original implementation. But that doesn't totally solve the issue because this movement is intense when you put several entities simultaneously.

Move, getCubes, t_ (tick), checkBlockCollisions, raytracing (double in interactions).

I do it privately at the moment, maybe I will share something other than advices and feedback sometime. 🚀

**Comment by Rastrian** [comment]
Bump @Lucaskyy

**Comment by Lucaskyy** [comment]
All good then, thank you for this PR! :)

**Pull request merged**